.github/workflows/s3-tests.yml

```diff
@@ -0,0 +1,318 @@
+        run: |
+          make prepare-test-env
+          echo "$(pwd)/vendor" >> $GITHUB_PATH
+        working-directory: neofs-dev-env
+
+      - name: Set zero fee
+        run: |
+          make update.container_fee val=0 && make update.container_alias_fee val=0
+        working-directory: neofs-dev-env
+
+      - name: Copy neofs-s3-gw binaries to vendor directory
+        run: |
+          cp bin/* ${GITHUB_WORKSPACE}/neofs-dev-env/vendor/
+        working-directory: neofs-s3-gw
+
+################################################################
+
+      - name: Log environment
+        run: |
+          echo "Check free space"
+          df -h
+          echo "=========================================="
+
+          echo "Check /etc/hosts"
+          cat /etc/hosts
+          echo "=========================================="
+
+          echo "Check docker images"
+          docker images
+          echo "=========================================="
+
+          echo "Check docker ps"
+          docker ps
+          echo "=========================================="
+
+          echo "Check neo-go version"
+          neo-go --version
+          echo "=========================================="
+
+          echo "Check neofs-s3-authmate version"
+          neofs-s3-authmate --version
+          echo "=========================================="
+
+          echo "Check neofs-s3-gw version"
+          echo "=========================================="
+          neofs-s3-gw --version
+          echo "=========================================="
+
+          echo "Check neofs-adm version"
+          neofs-adm --version
+          echo "=========================================="
+
+          echo "Check neofs-cli version"
+          neofs-cli --version
+          echo "=========================================="
+
+          echo "Check vendor dir"
+          ls -lah "${GITHUB_WORKSPACE}/neofs-dev-env/vendor"
+          echo "=========================================="
+
+        working-directory: neofs-dev-env
+
+################################################################
+
+      - name: Copy test-env file to .env for neofs-s3-dev-env
+        run: |
+          cp .env ${GITHUB_WORKSPACE}/neofs-s3-dev-env/.env
+        working-directory: neofs-dev-env
+
+      - name: Copy neofs-s3-authmate for neofs-s3-dev-env
+        run: |
+          cp ${GITHUB_WORKSPACE}/neofs-dev-env/vendor/neofs-s3-authmate bin/s3-authmate
+        working-directory: neofs-s3-dev-env
+
+      - name: Prepare test S3-gw test env
+        run: |
+          make prepare.s3-gw
+        working-directory: neofs-s3-dev-env
+
+      - name: Copy s3tests config to s3-tests directory
+        run: |
+          cp services/s3-gw/s3tests.conf ${GITHUB_WORKSPACE}/s3-tests/${S3_TESTS_CONFIG}
+        working-directory: neofs-s3-dev-env
+
+      - name: Prepare s3tests config
+        run: |
+          sed -i -e "s/host =.*$/host = ${S3_GW_IP}/" ${S3_TESTS_CONFIG}
```

Comment:
Dangerous regular expression - maybe there's more than one string that matches the `host =.*` pattern

Reply:
By default, sed uses only single-line matching. To enable multiline matching you need to enable it with 'N' command, like this:
`sed 'N; s/.* MULTILINE TEXT.*//' test.file`

So, greedy regex is pretty safe here. 


But, ok. I switched to a non-greedy '.*?' pattern

Reply:
It is best to replace lines like this with:
`sed -i -E "s/host =.*?$/host = ${S3_GW_IP}/" ${S3_TESTS_CONFIG}`
with the following:
`sed -i -E "s/host =__HOSTS_TO_CHANGE__/host = ${S3_GW_IP}/" ${S3_TESTS_CONFIG}`
Of course, this will require changes to the config file too.
That way you can be sure that some other line that someone else put in the config file won't change.
It will also make it easier for other users to interact with the config file.

Reply:
Go rid of this sed substitution with suitable employment of .env file for s3-dev-env.